src/map/entities/charentity.cpp

```diff
@@ -1077,6 +1077,80 @@ void CCharEntity::OnCastFinished(CMagicState& state, action_t& action)
+                        break;
+                    case SPELLFAMILY_WATER:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 5, 0, 0);
+                        break;
+                    case SPELLFAMILY_HYDROHELIX:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 5, 0, 0);
+                        break;
+                    case SPELLFAMILY_AERO:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 6, 0, 0);
+                        break;
+                    case SPELLFAMILY_ANEMOHELIX:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 6, 0, 0);
+                        break;
+                    case SPELLFAMILY_FIRE:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 3, 0, 0);
+                        break;
+                    case SPELLFAMILY_PYROHELIX:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 3, 0, 0);
+                        break;
+                    case SPELLFAMILY_BLIZZARD:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 7, 0, 0);
+                        break;
+                    case SPELLFAMILY_CRYOHELIX:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 7, 0, 0);
+                        break;
+                    case SPELLFAMILY_THUNDER:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 8, 0, 0);
+                        break;
+                    case SPELLFAMILY_IONOHELIX:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 8, 0, 0);
+                        break;
+                    case SPELLFAMILY_LUMINOHELIX:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 1, 0, 0);
+                        break;
+                    case SPELLFAMILY_NOCTOHELIX:
+                        effect = battleutils::GetSkillChainEffect(PTarget, 2, 0, 0);
+                        break;
+                    default:
+                        break;
+                }
+
+                if (effect != SUBEFFECT_NONE)
+                {
+                    int32 skillChainDamage = battleutils::TakeSkillchainDamage(static_cast<CBattleEntity*>(this), PTarget, actionTarget.param, nullptr);
+
+                    if (skillChainDamage < 0)
+                    {
+                        actionTarget.addEffectParam   = -skillChainDamage;
+                        actionTarget.addEffectMessage = 384 + effect;
+                    }
+                    else
+                    {
+                        actionTarget.addEffectParam   = skillChainDamage;
+                        actionTarget.addEffectMessage = 287 + effect;
+                    }
+                    actionTarget.additionalEffect = effect;
+                }
+
+                StatusEffectContainer->DelStatusEffectSilent(EFFECT_IMMANENCE);
```

Comment:
If no skillchain is possible at all (e.g. bio/drain) does it still delete immanence?

Reply:
I'm reaching out to some retail capture guys to try to get this detail confirmed.

Reply:
> I'm reaching out to some retail capture guys to try to get this detail confirmed.

do you still need any help with this? we're really interested in this functionality. i have access to retail. let me know.

Reply:
> > I'm reaching out to some retail capture guys to try to get this detail confirmed.
> 
> do you still need any help with this? we're really interested in this functionality. i have access to retail. let me know.

I requested details in captures discord but this hasn’t been confirmed yet. 
https://discord.com/channels/443544205206355968/443893540922064896/1201924153473507369

if you can confirm let me know and I’ll delete the request so siknoz doesn’t go for it.

Reply:
![image](https://github.com/LandSandBoat/server/assets/60417494/bd3d9358-550e-4782-9fb5-696480cc0f36)
Must be elemental magic it seems.

Reply:
Whoops, cropped that badly, 
![image](https://github.com/LandSandBoat/server/assets/60417494/7d0e4d3a-d78e-4e0e-b4cf-db88bab2b544)

Reply:
In hindsight, this should've been clear (scholar guides tell you to close Gravitation with Noctohelix and not Bio which is much faster)

Reply:
> Whoops, cropped that badly, ![image](https://private-user-images.githubusercontent.com/60417494/306530278-7d0e4d3a-d78e-4e0e-b4cf-db88bab2b544.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1MzQyNjgsIm5iZiI6MTcwODUzMzk2OCwicGF0aCI6Ii82MDQxNzQ5NC8zMDY1MzAyNzgtN2QwZTRkM2EtZDc4ZS00ZTBlLWI0Y2YtZGI4OGJhYjJiNTQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIxVDE2NDYwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc5ZDA0ZmI4NTA1Mzk5ZDlhZDdkODBhZTUzMDBiZDBkY2U5ODFjNmY4ZDIxNDVjNTI5NWI4MDU3ZWIwODkwYmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rw-wUMd_pYDGe0x-TkwM1CfgLJadp4hKdS5zNpqNzoY)

@WinterSolstice8, from what I am seeing your Immanence perpetuated through Bio, so it only deletes when there is a skillchain property triggered? I am updating this today and wanted to be sure were in agreement there.

Reply:
That is correct